src/keeper/contracts/ContractBase.ts

```diff
@@ -99,12 +101,16 @@ export abstract class ContractBase extends Instantiable {
         const method = this.contract.methods[name]
         try {
             const tx = method(...args)
-            const gas = await tx.estimateGas(args, {
-                from
+            let gas = await tx.estimateGas(args, {
+                from,
+                value
             })
 
+            if (value) gas += 21500
```

Comment:
should we make this a config value?

Reply:
yes that makes sense, it might be different on different chains